cmd/backup/main.go

```diff
@@ -93,17 +96,66 @@ type script struct {
 	webdavClient *gowebdav.Client
 	logger       *logrus.Logger
 	sender       *router.ServiceRouter
+	template     *template.Template
 	hooks        []hook
 	hookLevel    hookLevel
 
-	start  time.Time
-	file   string
-	output *bytes.Buffer
+	file  string
+	stats *Stats
 
-	c *config
+	c *Config
 }
 
-type config struct {
+// ContainersStats stats about the docker containers
+type ContainersStats struct {
+	All        int
+	ToStop     int
+	Stopped    int
+	StopErrors int
+}
+
+// BackupFileStats stats about the created backup file
+type BackupFileStats struct {
+	Name      string
+	FullPath  string
+	SizeBytes int64
+	SizeBin   string
+	SizeDec   string
+}
+
+// ArchiveStats stats about the status of an archival directory
+type ArchiveStats struct {
+	Total       int
+	Pruned      int
+	PruneErrors int
+}
+
+// ArchivesStats stats about each possible archival location (Local, WebDAV, S3)
+type ArchivesStats struct {
+	Local  ArchiveStats
+	WebDAV ArchiveStats
+	S3     ArchiveStats
+}
+
+// Stats global stats regarding script execution
+type Stats struct {
+	StartTime  time.Time
+	EndTime    time.Time
+	TookTime   time.Duration
+	Output     *bytes.Buffer
```

Comment:
Nit: maybe `LogOutput` would be a bit clearer, now that this has so many siblings.

Reply:
Agreed